src/Apps/CollectorProfile/Routes/Saves2/CollectorProfileSaves2Route.tsx

```diff
@@ -39,8 +48,30 @@ const CollectorProfileSaves2Route: FC<CollectorProfileSaves2RouteProps> = ({
   // Always display "Saved Artwork" collection first in the list
   const savedCollections = [savedCollection, ...otherCollections]
 
+  const onHistoryChanged = useCallback(
+    (id: string | null) => {
+      setActiveCollectionId(id ?? savedCollectionId)
+    },
+    [savedCollectionId, setActiveCollectionId]
+  )
+
+  useSavesHistoryEntity({
+    onChanged: onHistoryChanged,
+  })
+
+  const onDeleteCollection = () => {
+    setActiveCollectionId(savedCollectionId)
+    window.history.replaceState(null, "", "/collector-profile/saves2")
```

Comment:
Feel free to add any suggestions/comments. Especially on the use of `window.history`